manifests/daemon.pp

```diff
@@ -148,47 +155,60 @@
   if $manage_user {
     # if we manage the service, we need to reload it if our user changes
     # important for cases where another group gets added
-    if $manage_service {
+    if $manage_service and $real_service_ensure == 'running' {
       User[$user] ~> $notify_service
     }
     ensure_resource('user', [$user], {
-        ensure => 'present',
+        ensure => $ensure,
         system => true,
         groups => $extra_groups,
         shell  => $usershell,
     })
 
     if $manage_group {
-      Group[$group] -> User[$user]
+      if $ensure == 'present' {
+        Group[$group] -> User[$user]
+      } else {
+        User[$user] -> Group[$group]
+        Service[$name] -> User[$user]
+      }
     }
   }
   if $manage_group {
     ensure_resource('group', [$group], {
-        ensure => 'present',
+        ensure => $ensure,
         system => true,
     })
   }
 
   case $init_style { # lint:ignore:case_without_default
     'upstart': {
       file { "/etc/init/${name}.conf":
+        ensure  => stdlib::ensure($ensure, 'file'),
         mode    => '0444',
         owner   => 'root',
         group   => 'root',
         content => template('prometheus/daemon.upstart.erb'),
-        notify  => $notify_service,
       }
       file { "/etc/init.d/${name}":
-        ensure => link,
+        ensure => stdlib::ensure($ensure, 'file'),
         target => '/lib/init/upstart-job',
         owner  => 'root',
         group  => 'root',
         mode   => '0755',
       }
+      if $notify_service !~ Undef {
```

Comment:
```suggestion
      if $notify_service {
```

Reply:
I am not sure when a Type[Service] or Type[Exec] can be "falsy". So I would prefer to have this explicit